### DISCUSSION TITLE
How to replace strings in prerendered html?

### DISCUSSION THREAD

**Discussion by schaschko** [vikejs/vike#899] [Discussion] [answered]
With vite it's possible to replace strings in html by writing a plugin and insert it into `vite.config.ts`, e.g. `htmlPlugin` as shown below. In vite-plugin-ssr nothing happens, when using the following `vite.config.ts`.

I'm using `{ prerender: true }`. What is best practice to achieve that?

Background: I'm trying to insert `nonce` in script tags by using this approach.

```
import react from '@vitejs/plugin-react'

import ssr from 'vite-plugin-ssr/plugin'
import { UserConfig } from 'vite'

const htmlPlugin = () => {
  return {
    name: 'html-transform',
    transformIndexHtml(html) {
      return html.replace(
        /<title>(.*?)<\/title>/,
        `<title>Title replaced!</title>`,
      )
    },
  }
}

const config: UserConfig = {
  plugins: [react(), ssr({ prerender: true }), htmlPlugin()],
}

export default config
```

**Comment by brillout** [comment, marked as answer]
https://github.com/brillout/vite-plugin-ssr/issues/224
